src/distance.jl

```diff
@@ -1,192 +1,24 @@
-
-"""
-    diameter(eccentricities)
-    diameter(g, distmx=weights(g))
-
-Given a graph and optional distance matrix, or a vector of precomputed
-eccentricities, return the maximum eccentricity of the graph.
-
-# Examples
-```jldoctest
-julia> using LightGraphs
-
-julia> diameter(star_graph(5))
-2
-
-julia> diameter(path_graph(5))
-4
-```
-"""
-diameter(eccentricities::Vector) = maximum(eccentricities)
-diameter(g::AbstractGraph, distmx::AbstractMatrix=weights(g)) =
-    maximum(eccentricity(g, distmx))
-
-"""
-    periphery(eccentricities)
-    periphery(g, distmx=weights(g))
-
-Given a graph and optional distance matrix, or a vector of precomputed
-eccentricities, return the set of all vertices whose eccentricity is
-equal to the graph's diameter (that is, the set of vertices with the
-largest eccentricity).
-
-# Examples
-```jldoctest
-julia> using LightGraphs
-
-julia> periphery(star_graph(5))
-4-element Array{Int64,1}:
- 2
- 3
- 4
- 5
-
-julia> periphery(path_graph(5))
-2-element Array{Int64,1}:
- 1
- 5
-```
-"""
-function periphery(eccentricities::Vector)
-    diam = maximum(eccentricities)
-    return filter(x -> eccentricities[x] == diam, 1:length(eccentricities))
+function periphery(x...)
+    Base.depwarn("`periphery` is deprecated. Equivalent functionality has been moved to `LightGraphs.Measurements.periphery`.", :periphery)
+    LightGraphs.Measurements.periphery(x...)
 end
 
-periphery(g::AbstractGraph, distmx::AbstractMatrix=weights(g)) =
-    periphery(eccentricity(g, distmx))
-
-"""
-    radius(eccentricities)
-    radius(g, distmx=weights(g))
-
-Given a graph and optional distance matrix, or a vector of precomputed
-eccentricities, return the minimum eccentricity of the graph.
-
-# Examples
-```jldoctest
-julia> using LightGraphs
-
-julia> radius(star_graph(5))
-1
-
-julia> radius(path_graph(5))
-2
-```
-"""
-radius(eccentricities::Vector) = minimum(eccentricities)
-radius(g::AbstractGraph, distmx::AbstractMatrix=weights(g)) =
-    minimum(eccentricity(g, distmx))
-
-"""
-    center(eccentricities)
-    center(g, distmx=weights(g))
-
-Given a graph and optional distance matrix, or a vector of precomputed
-eccentricities, return the set of all vertices whose eccentricity is equal
-to the graph's radius (that is, the set of vertices with the smallest eccentricity).
-
-# Examples
-```jldoctest
-julia> using LightGraphs
-
-julia> center(star_graph(5))
-1-element Array{Int64,1}:
- 1
-
-julia> center(path_graph(5))
-1-element Array{Int64,1}:
- 3
-```
-"""
-function center(eccentricities::Vector)
-    rad = radius(eccentricities)
-    return filter(x -> eccentricities[x] == rad, 1:length(eccentricities))
+function radius(x...)
+    Base.depwarn("`radius` is deprecated. Equivalent functionality has been moved to `LightGraphs.Measurements.radius`.", :radius)
+    LightGraphs.Measurements.radius(x...)
 end
 
-center(g::AbstractGraph, distmx::AbstractMatrix=weights(g)) =
-    center(eccentricity(g, distmx))
+function center(x...)
```

Comment:
`@deprecate center(...) LightGraphs.Measurements.center(x...)`?

Reply:
I made the decision to use `depwarn` instead of `@deprecate` to keep things consistent with the rest of the changes, where `@deprecate` won't work.